File: client/src/app/domain/entity/Movie.ts
import {Category} from "./Category";

export class Movie {
  id?: number;
  title: string;
  price: number;
  categories: Category[];
  productionYear: number;
  description: string;

  constructor(id: number, title: string, price: number, categories: Category[], productionYear: number, description: string) {
    this.id = id;
    this.title = title;
    this.price = price;
    this.categories = categories;
    this.productionYear = productionYear;
    this.description = description;
  }
}

File: takeApi/src/main/java/poznan/take/domain/movie/entity/Movie.java
package poznan.take.domain.movie.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Movie {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String title;

    private double price;

    @ElementCollection(targetClass = Category.class)
    @Enumerated(EnumType.STRING)
    @CollectionTable
    private List<Category> categories;

    private int productionYear;

    private String description;
}